src/main/java/kubatech/tileentity/gregtech/multiblock/GT_MetaTileEntity_MegaIndustrialApiary.java

```diff
@@ -943,17 +943,20 @@ private Widget createBeesContainerWidget(EntityPlayer player) {
                         return;
                     }
                 }
-                if (clickData.mouseButton == 1) {
-                    if (player.inventory.getItemStack() == null) {
-                        player.inventory.setItemStack(removed.queenStack);
-                        ((EntityPlayerMP) player).isChangingQuantityOnly = false;
-                        ((EntityPlayerMP) player).updateHeldItem();
+                if (clickData.ctrl) {
+                    if (!addOutput(removed.queenStack)) {
+                        mStorage.add(removed);
```

Comment:
Add back where it was taken
```suggestion
                        mStorage.add(realID, removed);
```